### ISSUE TITLE
Multiple includes with a 1:1 association and M:N association.

### ISSUE THREAD

**Issue by johngiudici** [dchester/epilogue#58]
Consider the following test (sorry, it's huge, but it's very specific):

```
describe('Multiple Includes', function() {
  before(function() {
    test.models.Address = test.db.define('addresses', {
      id: { type: test.Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
      street: { type: test.Sequelize.STRING }
    }, {
      underscored: true,
      timestamps: false
    });

    test.models.Person = test.db.define('person', {
      id: { type: test.Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
      name: { type: test.Sequelize.STRING, unique: true }
    }, {
      underscored: true,
      timestamps: false
    });

    test.models.Hobby = test.db.define('hobby', {
      id: { type: test.Sequelize.INTEGER, autoIncrement: true, primaryKey: true },
      name: { type: test.Sequelize.STRING }
    }, {
      underscored: true,
      timestamps: false
    });

    test.models.Person.belongsTo(test.models.Address, {
      as: 'addy',
      foreignKey: 'addy_id'
    });
    test.models.Person.belongsToMany(test.models.Hobby, {
      as: 'hobbies',
      through: 'person_hobbies'
    });
    test.models.Hobby.belongsToMany(test.models.Person, {
      as: 'people',
      through: 'person_hobbies'
    });
  });

  beforeEach(function(done) {
    test.initializeDatabase(function() {
      test.initializeServer(function() {
        rest.initialize({
          app: test.app,
          sequelize: test.Sequelize
        });

        rest.resource({
          model: test.models.Person,
          include: [
            { model: test.models.Address, as: 'addy' },
            { model: test.models.Hobby, as: 'hobbies' }
          ],
          endpoints: ['/personWithTwoIncludes', '/personWithTwoIncludes/:id']
        });

        done();
      });
    });
  });
  
  it('should be ok', function(done) {
    test.models.Person.create({ name: 'testName' })
      .then(function() {
        request.get({
          url: test.baseUrl + '/personWithTwoIncludes'
        }, function(error, response, body) {
          expect(response.statusCode).to.equal(200);
          done();
        });
      });
  });
});
```

The body returned by the request is:
`{"message":"internal error","errors":["SQLITE_ERROR: no such column: person.addy_id"]}`

This is because we filter out related fields in our attributes list inside the list controller. It is only a problem in this scenario because sequelize generates a subquery when it has a M:N relationship and a limit, and us omitting `addy_id` from our attributes confuses it. The easiest solution is to not omit associated fields, but then we end up with extraneous data in our response body (here, we'd have a `body.Addy` object as well as a `body.addy_id` field). 

Thoughts?

**Comment by mbroadst** [comment]
Off the top of my head I'd say we can solve this by removing the include attributes ourselves after the findAll, it would be a regression performance-wise, but would solve your particular case.

Another option is to investigate ways to build this kind of behavior into the "auto association" code in my branch over [here](https://github.com/dchester/epilogue/tree/auto-associations). Perhaps we could do some up front inspection of the models and maintain two lists: includeAttributes (to be removed from the attributes selected for), and removeAfterFindAttributes which are keys that aren't passed into the findAll attributes but removed after the search is performed. In fact, you wouldn't really need to mess with the auto-associations branch to do just that I just want someone to play with that code :smile: 

Yeah so I guess my thoughts are: if its not too crazy, do the second thing I suggested there (create an extra array of attributes to be removed after based on their existence as foreign keys). Whatcha think?

**Comment by johngiudici** [comment]
I've added a change that filters out attributes after the find. I don't think it will be too much of a performance hit since we have a hard limit of 1000 records, but I could write a timing test if anyone wanted it. One thing we may have to look at in the future is that we don't filter nested associations. In my PR, this means that the `hobby` record includes `person_hobbies`.

**Comment by mbroadst** [comment]
@johngiudici I think this is closed with #59 right?

**Comment by johngiudici** [comment]
Yes, it is. I still can't seem to figure out how to link a new PR with an existing issue XD
